ships/0005-local-source-upload.md

```diff
@@ -0,0 +1,250 @@
+---
+title: local-source-upload
+authors:
+  - @otaviof
+reviewers:
+  - @alicerum
+  - @adambkaplan
+approvers:
+creation-date: 2021-06-23
+last-updated: 2021-06-23
+---
+
+# Source Upload
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [docs](/docs/)
+
+## Summary
+
+Proposal to improve developer-experience by allowing to upload a local repository clone to the
+Kubernetes cluster, and use Shipwright Build Controller to produce a container-image out of it.
+
+As a developer, I can try Shipwright before opening a pull-request, and take advantage of the cluster
+resources to create the image. Effectively moving shipwright closer to the developer's inner-loop, in
```

Comment:
should we assume everyone knows what "inner-loop" means in this context?

For example, in a broader sw sense, you have https://en.wikipedia.org/wiki/Inner_loop

Conversely, in the openshift context, there are public articles like https://www.openshift.com/blog/openshift-4-2-expanded-tools-and-services-for-developers and https://developers.redhat.com/blog/2020/06/16/enterprise-kubernetes-development-with-odo-the-cli-tool-for-developers out there

perhaps a quick definition or reference link here.

Reply:
Indeed, good suggestion! I'll amend with the wikipedia link.

Reply:
actually the wikipedia link is the wrong definition for our purposes :-)

sorry for the confusion, I was trying to illustrate the varying definitions

the other 2 redhat/openshift links get into the inner loop we are talking about here

Reply:
Ohh, I see. Thanks for the heads up, right on time :-)

Reply:
Please consider the new way I'm wording it, let me know if makes sense.

Reply:
looks good feel free to mark this one resolved